lib/prism/translation/parser/lexer.rb

```diff
@@ -252,18 +254,19 @@ def to_a
           while index < length
             token, state = lexed[index]
             index += 1
-            next if %i[IGNORED_NEWLINE __END__ EOF].include?(token.type)
+            next if TYPES_ALWAYS_SKIP.include?(token.type)
 
             type = TYPES.fetch(token.type)
             value = token.value
             location = range(token.location.start_offset, token.location.end_offset)
 
             case type
             when :kDO
-              types = tokens.map(&:first)
-              nearest_lambda_token_type = types.reverse.find { |type| LAMBDA_TOKEN_TYPES.include?(type) }
+              nearest_lambda_token = tokens.reverse_each.find do |token|
+                LAMBDA_TOKEN_TYPES.include?(token[0])
+              end
 
-              if nearest_lambda_token_type == :tLAMBDA
+              if nearest_lambda_token && nearest_lambda_token[0] == :tLAMBDA
```

Comment:
It can be made slightly less redundant.

```suggestion
              if nearest_lambda_token&.first == :tLAMBDA
```